### PR TITLE
Add: integrate GitHub data handling and improve error logging; update…

### PR DESCRIPTION
**Pull request by AkshatRana09**
# Integrate GitHub scraper, improve error logging, and update dependencies

## Description
Add GitHub data ingestion and normalization to the platform pipeline, improve server error logging for clearer diagnostics, and update project dependencies.

## Changes
- Add GitHub scraper to fetch and normalize profile/repo metrics.
- Incorporate GitHub data into platform aggregation logic.
- Improve server error logging (more informative messages and stack traces).
- Update dependencies in package.json.

## Why
Enables GitHub as an additional data source for platform metrics and improves observability and maintainability via clearer logs and dependency updates.

## Testing / Review notes
- Verify platform endpoints return expected GitHub-related fields without breaking existing platforms.
- Test edge cases in the GitHub scraper: rate limits, missing fields, partial responses.
- Run existing tests and exercise API endpoints (e.g., GET /platforms).

## Suggested reviewer checklist
- Confirm mapping logic in platform controller handles missing GitHub fields safely.
- Verify new logging does not leak sensitive data.
- Run dependency audit and smoke test after package changes.

### CODE REVIEW

File: src/scrapers/github.scraper.js [hunk 1]
```diff
@@ -180,8 +190,12 @@ async function fetchGitHubData(username) {
       })),
     };
   } catch (err) {
-    return { error: "Internal server error", detail: err.message };
+    console.error("❌ GitHub API Request Failed:", err.response?.data || err.message);
```
Comment (mdmokaramkhan) [inline src/scrapers/github.scraper.js]: we have a built in logger use that in proper logging format

Comment (mdmokaramkhan) [reply src/scrapers/github.scraper.js]: Ok i fixed it

Comment (AkshatRana09) [reply src/scrapers/github.scraper.js]: okay

**Review by mdmokaramkhan — COMMENTED** [review]
Done github
